src/safe_cli/tx_builder/tx_builder_file_decoder.py

```diff
@@ -0,0 +1,240 @@
+    contract_fields = contract_method.get("inputs", []) if contract_method else []
+
+    is_valid_contract_method = (
+        contract_method_name and contract_method_name not in NON_VALID_CONTRACT_METHODS
+    )
+
+    if is_valid_contract_method:
+        try:
+            encoding_types = _parse_types_to_encoding_types(contract_fields)
+            values = [
+                parse_input_value(
+                    field["type"], contract_fields_values.get(field["name"], "")
+                )
+                for field in contract_fields
+            ]
+
+            function_signature = f"{contract_method_name}({','.join(encoding_types)})"
+            function_selector = Web3.keccak(text=function_signature)[:4]
+            encoded_parameters = encode_abi(encoding_types, values)
+            hex_encoded_data = HexBytes(function_selector + encoded_parameters)
+            return hex_encoded_data
+        except Exception as error:
+            raise TxBuilderEncodingError(
+                "Error encoding current form values to hex data:", error
+            )
+    else:
+        raise InvalidContratMethodError(
+            f"Invalid contract method {contract_method_name}"
+        )
+
+
+def parse_boolean_value(value: str) -> bool:
+    if isinstance(value, str):
+        if value.strip().lower() in ["true", "1"]:
+            return True
+
+        if value.strip().lower() in ["false", "0"]:
+            return False
+
+        raise SoliditySyntaxError("Invalid Boolean value")
+
+    return bool(value)
+
+
+def parse_int_value(value: str) -> int:
+    trimmed_value = value.replace('"', "").replace("'", "").strip()
+
+    if trimmed_value == "":
+        raise SoliditySyntaxError("Invalid empty strings for integers")
+    try:
+        if not trimmed_value.isdigit() and bool(
+            re.fullmatch(r"0[xX][0-9a-fA-F]+|[0-9a-fA-F]+$", trimmed_value)
+        ):
+            return int(trimmed_value, 16)
+
+        return int(trimmed_value)
+    except ValueError:
+        raise SoliditySyntaxError("Invalid integer value")
+
+
+def parse_string_to_array(value: str) -> List[Any]:
+    number_of_items = 0
+    number_of_other_arrays = 0
+    result = []
+    value = value.strip()[1:-1]  # remove the first "[" and the last "]"
+
+    for char in value:
+        if char == "," and number_of_other_arrays == 0:
+            number_of_items += 1
+            continue
+
+        if char == "[":
+            number_of_other_arrays += 1
+        elif char == "]":
+            number_of_other_arrays -= 1
+
+        if len(result) <= number_of_items:
+            result.append("")
+
+        result[number_of_items] += char.strip()
+
+    return result
+
+
+def _get_base_field_type(field_type: str) -> str:
+    trimmed_value = field_type.strip()
+    if not trimmed_value:
+        raise SoliditySyntaxError("Empty base field type for")
+
+    base_field_type_regex = re.compile(
+        r"^([a-zA-Z0-9]*)(((\[\])|(\[[1-9]+[0-9]*\]))*)?$"
```

Comment:
There's no need to escape the `]` here.

```suggestion
        r"^([a-zA-Z0-9]*)(((\[])|(\[[1-9]+[0-9]*]))*)?$"
```